phoenicis-tools/src/main/java/org/phoenicis/tools/system/ScreenManager.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2015-2017 PÂRIS Quentin
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package org.phoenicis.tools.system;
+
+import org.phoenicis.configuration.security.Safe;
+import org.phoenicis.entities.OperatingSystem;
+
+public class setVirtualDesktop {
+    /**
+     *Fetch the screen resolution and sets the virtual desktop to match it
+     *
+     *@return the current screen resolution
+     */
+    public virtualDesktop fetchCurrentScreenResolution() {
```

Comment:
`getScreenSize()` returns a `Dimension` object, see also https://docs.oracle.com/javase/9/docs/api/java/awt/Toolkit.html#getScreenSize--

Reply:
Should it be converted to text?

Reply:
What do you mean by `to text`?

Reply:
Something like `java.awt.Toolkit.getDefaultToolkit().getScreenSize().toText()` because a `Dimension` object won't be recorginzed by Wine (at least that's what I think)

Reply:
In which format is the screen size needed in the scripts? Is it always the same?

Reply:
I suggest two methods for x and y resolution returning `int`.

Reply:
I suggest creating a Resolution object, may be easier to use after

Reply:
Or getting the x and y from an already existing `Dimension` object.

Reply:
In general, we try to avoid that because you cannot exactly control what can Dimension do

Reply:
And we do not want to expose standard Java classes by default. So yes, either the `Resolution` object or x/y to avoid a separate class.

Reply:
@madoar the format would be screenWidth x screenHeight, always the same

Reply:
I think I prefer a separate class, this reduces the number of necessary calls from JS to Java.

Reply:
@Zemogiter you need to change the the line above to `public Dimension fetchCurrentScreenResolution() {` to make it work